src/transformStateWithClones.js

```diff
@@ -7,7 +9,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const history = [];
+  let current = { ...state };
+
+  for (const action of actions) {
+    let newCurrent = { ...current };
```

Comment:
The creation of `newCurrent` here is redundant since it is immediately overwritten in the switch cases. Consider initializing `newCurrent` only when needed within each case to improve efficiency.